src/js/state/History/types.js

```diff
@@ -1,11 +1,12 @@
 /* @flow */
 
-import type {SearchRecord} from "../../types"
+import type {ScrollPosition, SearchRecord} from "../../types"
 import type {Ts} from "../../brim"
 
 export type HistoryState = {
   position: number,
-  entries: SearchRecord[]
+  entries: SearchRecord[],
+  scrollPositions: ScrollPosition[]
```

Comment:
As mentioned earlier, I think it would be better to add a `scrollPos` property to the `SearchRecord` objects, instead of adding another array here.